### DISCUSSION TITLE
Laser support with and Lightburn

### DISCUSSION THREAD

**Discussion by Cncninja** [terjeio/grblHAL#208] [Q&A]
I have the Teensy4.1 and Phil board, I was going to put it on a 3018 mill but decided to use it on my China 60w laser instead after see the light burn software, but im a bit lost i dont see much about laser support. tho I did see someone had it running but had to just post-process to gcode and use I/O sender 8 v.s direct control. is there any documentation for compiling for a laser and what pins for laser control?

Thank you

**Comment by phil-barrett** [comment]
You don't need to compile for laser - the default build supports it. The old 8-bit Grbl needed a special build but grblHAL added it into the base software. Laser mode works as a Grbl setting ($32=1).  Grbl uses pretty standard GCode for lasers. If your laser controller takes 0-10V control voltage you would use the 0-10V output for laser intensity control.  If it takes a PWM, you would use the PWM output. Laser output in GCode is controlled via M3 (CW) or M4 (CCW/ACW). Sometimes a CCW direction is used by laser SW (Dir on the pin header or screw terminal, upper left on the PCB). Enable pin being high (EN on the pin header or screw terminal) will always indicate laser power. For percent intensity, the S param in GCode is used.  Leaving the Grbl Maximum Spindle Speed setting ($30) at 1000, the percent power is the S-value divided by 1000. An S500 would give a %50 (500/1000) intensity.

Lightburn works with grblHAL direct though I have not personally used it.

**Comment by terjeio** [comment]
@Cncninja If you (want to) do raster engraving with lots of small moves at high speed then a bit of tuning may be help. See issue #142 for a discussion related to that.
